### PR TITLE
gh-108494: Document how to add a project in PCbuild/readme.txt

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-108494

### CODE REVIEW

**Comment by vstinner** [comment]
cc @zooba @zware

File: PCbuild/readme.txt [hunk 1]
```diff
@@ -293,3 +293,23 @@ project, with some projects overriding certain specific values. The GUI
...
 with false information, especially for settings that automatically adapt
 for different configurations.
+
+Add a new project
+-----------------
+
+For example, add a new _testclinic_limited project to build a new
+_testclinic_limited extension, the file Modules/_testclinic_limited.c:
+
+* In PCbuild/, copy _asyncio.vcxproj to _testclinic_limited.vcxproj,
```
Comment (zooba) [inline PCbuild/readme.txt]: I'd suggest copying an existing `test` project if you're adding a test project, or copy an existing library module if you're adding a library module.

The main difference is whether they get PGO or not, but also whether a PGO failure will break the entire build.

File: PCbuild/readme.txt [hunk 2]
```diff
@@ -293,3 +293,23 @@ project, with some projects overriding certain specific values. The GUI
...
+  replace RootNamespace value with `_testclinic_limited`, replace
+  `_asyncio.c` with `_testclinic_limited.c`.
+* Open Visual Studio, open PCbuild\pcbuild.sln solution, add the
+  PCbuild\_testclinic_limited.vcxproj project to the solution ("add existing
+  project).
+* Add a dependency on the python project to the new _testclinic_limited
+  project.
+* Save and exit Visual Studio.
+* Add `;_testclinic_limited` to `<TestModules Include="...">` in
+  PCbuild\pcbuild.proj.
```
Comment (zooba) [inline PCbuild/readme.txt:313]: Very likely that `Tools\msi\lib\lib_files.wxs` and/or `Tools\msi\test\test_files.wxs` will also need updating to get it into the installer.

`PC\layout\main.py` only needs updating if you add a test-only extension whose name doesn't start with `_test`.

File: PCbuild/readme.txt [hunk 3]
```diff
@@ -293,3 +293,23 @@ project, with some projects overriding certain specific values. The GUI
...
+* Build Python from scratch (clean the solution) to check that the new project
+  is built successfully.
```
Comment (zooba) [inline PCbuild/readme.txt:320]:
```suggestion
  is built successfully.
* Ensure the new .vcxproj and .vcxproj.filters files are added to your commit,
  as well as the changes to pcbuild.sln, pcbuild.proj and any other modified
  files.
```

**Review by zooba — COMMENTED** [review]
Thanks for starting this!

We should probably eventually add how to do it by hand, as I think that's how most contributors add them, but having this much here will hopefully convince someone else to write it up.

**Comment by vstinner** [comment]
@zooba: I updated the PR to address your review.

**Comment by zooba** [comment]
LGTM 👍

**Comment by vstinner** [comment]
More doc is always a good think, thanks for the review. I didn't know the parts that you showed me.

**Pull request merged**